### PR TITLE
Fix Symfony deprecations 

### LINKED ISSUE #12551: Check & fix Symfony deprecations
### Improvement description

There are still some deprecation messages triggered when running the Codeception tests. We should try to get rid of them in 10.5. 

```
DEPRECATION: Since symfony/framework-bundle 5.3: The "session" service and "SessionInterface" alias are deprecated, use "$requestStack->getSession()" instead. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/framework-bundle 5.3: The "session.storage.mock_file" service is deprecated, use "session.storage.factory.mock_file" instead. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/framework-bundle 5.3: The "session.storage.metadata_bag" service is deprecated, create your own "session.storage.factory" instead. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/framework-bundle 5.1: The "session.flash_bag" service is deprecated, use "$session->getFlashBag()" instead. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/event-dispatcher 5.1: Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy is deprecated, use the event dispatcher without the proxy. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/security-core 5.4: The $autenticationManager argument of "Symfony\Component\Security\Core\Authorization\AuthorizationChecker::__construct" is deprecated. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25       
DEPRECATION: Since symfony/lock 5.4: Usage of a DBAL Connection with "Symfony\Component\Lock\Store\PdoStore" is deprecated and 
will be removed in symfony 6.0. Use "Symfony\Component\Lock\Store\DoctrineDbalStore" instead. /var/www/html/vendor/symfony/contracts/Deprecation/function.php:25

DEPRECATION: Since symfony/event-dispatcher 5.1: Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy is deprecated, use the event dispatcher without the proxy./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/security-core 5.4: The $autenticationManager argument of "Symfony\Component\Security\Core\Authorization\AuthorizationChecker::__construct" is deprecated./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25
DEPRECATION: Since symfony/lock 5.4: Usage of a DBAL Connection with "Symfony\Component\Lock\Store\PdoStore" is deprecated and will be removed in symfony 6.0. Use "Symfony\Component\Lock\Store\DoctrineDbalStore" instead./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25

DEPRECATION: Since symfony/twig-bundle 5.2: Accessing the "twig" service directly from the container is deprecated, use dependency injection instead./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25

```

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #12551

### CODE REVIEW

File: bundles/AdminBundle/Security/User/UserProvider.php [hunk 1]
```diff
@@ -37,6 +37,16 @@ public function loadUserByUsername($username)
...
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * @deprecated use loadUserByIdentifier() instead.
+     */
+    public function loadUserByUsername($identifier)
+    {
+        return $this->loadUserByUsername($identifier);
```
Comment (kingjia90) [inline bundles/AdminBundle/Security/User/UserProvider.php]:
```suggestion
        return $this->loadUserByIdentifier($identifier);
```

File: bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $loader = new YamlFileLoader(
+            $container,
+            new FileLocator(__DIR__.'/../../Resources/config')
+        );
+
+        if (!$container->getParameter('security.authenticator.manager.enabled')) {
```
Comment (kingjia90) [inline bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php]: i get an error, reads as `You have requested a non-existent parameter "security.authenticator.manager.enabled".`
it's loading `LegacyAuthenticationSecurityPass` before `AuthenticatorSecurity`, so the line `$container->setParameter('security.authenticator.manager.enabled', true);` is not set yet?

Comment (kingjia90) [reply bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php:42]: even after the `has` change, it doesn't work, throws an exception here 
`Invalid service "Pimcore\Bundle\AdminBundle\Security\Guard\AdminAuthenticator": the class is not set.`

**Review by kingjia90 — CHANGES_REQUESTED** [review]

**Comment by kingjia90** [comment]
These 2 are still triggered in the tests: 
- [x] `Since symfony/twig-bundle 5.2: Accessing the "twig" service directly from the container is deprecated, use dependency injection instead./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25`
- [x] ~`Since symfony/event-dispatcher 5.1: Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy is deprecated, use the event dispatcher without the proxy./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25`~

are these missing here or already covered in another unmerged PR?

File: bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php (hunk 2 above)
Comment (dvesh3) [reply bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php]: Fixed it by by checking `hasParameter('security.authenticator.manager.enabled')`

Comment (kingjia90) [reply bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php]: I've also tried with hasParameter, but it throws another problem, related to the Service not set. Have you tried to log into admin interface?

Comment (dvesh3) [reply bundles/AdminBundle/DependencyInjection/Compiler/LegacyAuthenticationSecurityPass.php]: yes, I have fixed the service not set issue too!

File: bundles/AdminBundle/DependencyInjection/PimcoreAdminExtension.php [hunk 3]
```diff
@@ -52,6 +52,12 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('export.yaml');
         $loader->load('aliases.yaml');
 
+        //@TODO: Remove in Pimcore 11
+        if(!$container->hasParameter('pimcore_admin.security.logout_success_handler')) {
```
Comment (dvesh3) [inline bundles/AdminBundle/DependencyInjection/PimcoreAdminExtension.php]:
```suggestion
        if(!$container->hasParameter('security.authenticator.manager.enabled')) {
```

**Comment by dvesh3** [comment]
>Since symfony/event-dispatcher 5.1: Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy is deprecated, use the event dispatcher without the proxy./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25

@kingjia90  I'm not really sure, where this is coming from..we don't use this proxy class anywhere in the core or Pimcore bundles.

**Comment by kingjia90** [comment]
> > Since symfony/event-dispatcher 5.1: Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy is deprecated, use the event dispatcher without the proxy./home/runner/work/pimcore/pimcore/vendor/symfony/contracts/Deprecation/function.php:25
> 
> @kingjia90 I'm not really sure, where this is coming from..we don't use this proxy class anywhere in the core or Pimcore bundles.

Ok, it seems that some symfony components have a bc layer that reference this 
https://github.com/symfony/mailer/blob/57384499f838a00776fb1d9e58f7e6e1a706cf71/Mailer.php#L39
and probably it triggers the notice when composer autoloads (i can find LegacyEventDispatcherProxy in `/vendor/composer/autoload_classmap.php`)
https://github.com/symfony/event-dispatcher/blob/8e6ce1cc0279e3ff3c8ff0f43813bc88d21ca1bc/LegacyEventDispatcherProxy.php#L16

**Review by kingjia90 — APPROVED** [review]

**Pull request merged**